CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+# Contributing
+
+## Contributor's Agreement
+
+You are under no obligation whatsoever to provide any bug fixes,
+patches, or upgrades to the features, functionality or performance of
+the source code ("Enhancements") to anyone; however, if you choose to
+make your Enhancements available either publicly, or directly to the
+project, without imposing a separate written license agreement for
+such Enhancements, then you hereby grant the following license: a
+non-exclusive, royalty-free perpetual license to install, use, modify,
+prepare derivative works, incorporate into other computer software,
+distribute, and sublicense such enhancements or derivative works
+thereof, in binary and source code form.
+
+## [Technical Charter](https://apptainer.org/technical-charter)
+
+- section 2(b) requires a list of Technical Steering Committee (TSC)
+  members here
+
+### Technical Steering Committee
+
+(in alphabetical sort order of surname)
+
+```text
+- Jonathon Anderson <jonathon.anderson@ciq.co>, <anderbubble@gmail.com>
+- Christian Goll <christian.goll@googlemail.com>
+- John Hanks <griznog@gmail.com>
+- Gregory Kurtzer <gmkurtzer@gmail.com>
+- Jeremy Siadal <jeremy.c.siadal@intel.com>
+```
+
+## Getting Started
+
+When contributing to Warewulf, it is important to properly communicate
+the gist of the contribution. If it is a simple code or editorial fix,
+simply explaining this within the GitHub Pull Request (PR) will
+suffice. But if this is a larger fix or Enhancement, you are advised
+to first discuss the change with the project leader or developers.
+
+<!--
+Please note we have a [code of conduct](CODE_OF_CONDUCT.md). Please
+follow it in
+all your interactions with the project members and users.  
+
+In addition, all commits must be "Signed-off" by using `git commit
+-s`,
+acknowledging that you agree to the [Developer Certificate of Origin](DCO.md).  
+-->
+
+## Pull Requests (PRs)
+
+1. Essential bug fix PRs should be sent to both main and release
+   branches.
+2. Small bug fix and feature enhancement PRs should be sent to main
+   only.
+3. Follow the existing code style precedent. For Go, you will mostly
+   conform to the style and form enforced by the "go fmt" and "golint"
+   tools for proper formatting.
+4. For any new functionality, please write appropriate go tests that
```

Comment:
I would add, before adding new functionality, check in with the community for feedback on your idea before embarking on a path that would not be accepted into the code base.  Something like that.

I do see it in the Getting Started section. It might be beneficial to repeat that idea here as well.  To be honest, I glossed over that point in Getting Started and jumped straight to here at first.

Reply:
I just merged the Getting Started section into the PRs section, so there wouldn't be two places to look. ^_^